### PR TITLE
Feature/archive

### PR DESCRIPTION
**Pull request by Kevin-Solomon**
# Features added :
- [x] Add to archive feature.
- [x] Delete from archive feature.
- [x] Restore from archive feature.  
- [x] Sort By latest and oldest notes. ***the UI of the sort div will become better just worked on functionality as of now***
 
***I am aware that must shift the notecontainer and sidebar components to the component folder instead of having them in the home page folder***

### CODE REVIEW

File: src/components/Filter/Filter.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import { useFilter } from '../../context/filter/filterContext';
+import './Filter.css';
+function Filter() {
+  const [hidden, setHidden] = useState(true);
+  const { filterState, filterDispatch } = useFilter();
+  console.log(filterState);
```
Comment (VishalPatil18) [inline src/components/Filter/Filter.jsx]: You could remove the unwanted `console.log()`'s

File: src/components/Filter/Filter.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <div className="filter-container">
+      <button className="btn" onClick={() => setHidden(!hidden)}>
```
Comment (VishalPatil18) [inline src/components/Filter/Filter.jsx]: I think you should do functional updates in setState.
i.e.
```jsx
setHidden(prevState => !prevState)
```

File: src/components/Filter/Filter.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        Filter
+      </button>
+      <div className={hidden ? 'filter' : 'filter show'}>
+        <div>Sort By</div>
+        <div>
+          <input
+            id="latest-first"
+            type="radio"
+            checked={filterState.sortBy === 'LATEST_FIRST'}
+            onChange={e => filterDispatch({ type: 'LATEST_FIRST' })}
```
Comment (VishalPatil18) [inline src/components/Filter/Filter.jsx]: What is the need of passing `e` as an argument here? I think you could do it without that as well i.e. 
```jsx
onChange={()=> filterDispatch({type: "LATEST_FIRST"})}
```

File: src/pages/Home/Home.jsx [hunk 4]
```diff
@@ -3,13 +3,16 @@ import './Home.css';
 import { Navbar } from '../../components/Navbar/Navbar';
 import Sidebar from './components/Sidebar/Sidebar';
 import { NoteContainer } from './components/NoteContainer/NoteContainer';
+import { notesState, useNotes } from './../../context/notes/noteContext';
 export const Home = () => {
+  const { noteState } = useNotes();
+  console.log(noteState);
```
Comment (VishalPatil18) [inline src/pages/Home/Home.jsx]: You could remove the unwanted `console.log()`'s

File: src/pages/Home/components/NoteContainer/NoteContainer.jsx [hunk 5]
```diff
@@ -1,18 +1,36 @@
...
-import { useNotes } from './../../../../context/notes/noteContext';
-
-export const NoteContainer = () => {
-  const { noteState } = useNotes();
+import { getSortedList } from '../../../../util/getSortedList';
+import { useFilter } from '../../../../context/filter/filterContext';
+export const NoteContainer = ({ notes, inArchive }) => {
+  const { filterState } = useFilter();
+  const sortedList = getSortedList(filterState, notes);
+  console.log(sortedList);
```
Comment (VishalPatil18) [inline src/pages/Home/components/NoteContainer/NoteContainer.jsx]: You could remove the unwanted `console.log()`'s

File: src/reducer/noteReducer.js [hunk 6]
```diff
@@ -1,15 +1,26 @@
-const noteInitialState = { notes: [] };
+const noteInitialState = { notes: [], archives: [] };
 const ADD_NEW_NOTE = 'ADD_NEW_NOTE';
 const UPDATE_NOTE = 'UPDATE_NOTE';
 const DELETE_NOTE = 'DELETE_NOTE';
+const ARCHIVE_NOTE = 'ARCHIVE_NOTE';
+const RESTORE_ARCHIVE_NOTE = 'RESTORE_ARCHIVE_NOTE';
+const DELETE_ARCHIVE_NOTE = 'DELETE_ARCHIVE_NOTE';
 const noteReducer = (state, { type, payload }) => {
+  console.log('in dispatch');
```
Comment (VishalPatil18) [inline src/reducer/noteReducer.js]: You could remove the unwanted `console.log()`'s

File: src/reducer/noteReducer.js [hunk 7]
```diff
@@ -1,15 +1,26 @@
...
     case UPDATE_NOTE:
       return { ...state, notes: [...payload] };
     case DELETE_NOTE:
       return { ...state, notes: [...payload] };
+    case ARCHIVE_NOTE:
+      return { ...state, ...payload };
+    case RESTORE_ARCHIVE_NOTE:
+      return { ...state, ...payload };
+    case DELETE_ARCHIVE_NOTE:
+      console.log('in delete archive');
```
Comment (VishalPatil18) [inline src/reducer/noteReducer.js]: You could remove the unwanted `console.log()`'s

File: src/util/deleteArchiveNote.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import axios from 'axios';
+export const deleteArchiveNote = async (_id, token, noteDispatch) => {
+  console.log('in delete aarchive function');
```
Comment (VishalPatil18) [inline src/util/deleteArchiveNote.js]: You could remove the unwanted `console.log()`'s

File: src/util/deleteArchiveNote.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  const response = await axios({
+    method: 'delete',
+    url: `/api/archives/delete/${_id}`,
+    headers: { authorization: token },
+  });
+  console.log(response);
```
Comment (VishalPatil18) [inline src/util/deleteArchiveNote.js]: You could remove the unwanted `console.log()`'s

File: src/util/restoreArchiveNote.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import axios from 'axios';
+export const restoreArchiveNote = async (_id, token, noteDispatch) => {
+  try {
+    const response = await axios({
+      method: 'post',
+      url: `/api/archives/restore/${_id}`,
+      headers: { authorization: token },
+    });
+    console.log(response);
```
Comment (VishalPatil18) [inline src/util/restoreArchiveNote.js]: You could remove the unwanted `console.log()`'s

**Review by VishalPatil18 — APPROVED** [review]
Try avoiding unwanted `console.log()`'s in your PR.
The rest looks quite good. Great work @Kevin-Solomon 👍

**Review by AtulPant2704 — APPROVED** [review]
Code is well structured with proper naming just remove console.log from next time onwards. Rest looks good.

File: src/util/restoreArchiveNote.js (hunk 10 above)
Comment (Kevin-Solomon) [reply src/util/restoreArchiveNote.js]: Thank you Vishal for pointing out the unwanted ```console.log```.

**Pull request merged**